Add lesson standards via autocomplete dropdown

davidsbailey opened the pull request:
Continues [PLAT-328]. Depends on #39736, which adds backend support for searching for standards.

https://user-images.githubusercontent.com/8001765/112588466-afef5d80-8dbc-11eb-8d61-9f05fe20cf67.mov

## Testing story

New unit tests cover redux actions and lesson controller update. I'm open to ideas for how to test the dropdown functionality. for now I've just tested it manually.

## Follow-up work

remaining steps needed for April deadline are tracked in the [design doc](https://docs.google.com/document/d/1zvdv7y8VuRz8KqT1rzqeJAmvirCvUr1jS8OWy0p9FHk/edit#heading=h.hgiwmznxunug).

[PLAT-328]: https://codedotorg.atlassian.net/browse/PLAT-328

dmcavoy commented on apps/src/lib/levelbuilder/lesson-editor/StandardsEditor.jsx:
```diff
@@ -135,10 +145,70 @@ class StandardsEditor extends Component {
+          <strong>Select a Standard to add</strong>
+        </label>
+        <SearchBox
+          // Specify a key in order to force this component to remount when
```
Nice find!

davidsbailey replied:
Thanks! I did notice one related problem, which is that the logic for filtering out standards which are already in the lesson does not get updated after you add a standard. so, if you add a standard, then search for the same thing again, the standard you just added will still be in the list. https://codedotorg.atlassian.net/browse/PLAT-886

dmcavoy commented on apps/src/lib/levelbuilder/lesson-editor/StandardsEditor.jsx:
```diff
@@ -135,10 +145,70 @@ class StandardsEditor extends Component {
+        </label>
+        <select onChange={this.handleSelectFramework} style={styles.select}>
+          <option value="">(none)</option>
+          <option value="iste">ISTE Standards for Students</option>
```
Should this list be pulled from somewhere?

davidsbailey replied:
oops, yes: https://codedotorg.atlassian.net/browse/PLAT-885

dmcavoy commented on apps/src/lib/levelbuilder/lesson-editor/StandardsEditor.jsx:
```diff
@@ -135,10 +145,70 @@ class StandardsEditor extends Component {
+          // a query is repeated after changing the framework.
+          key={this.state.frameworkShortcode}
+          onSearchSelect={this.onSearchSelect}
+          searchUrl={'standards/search'}
```
Oh I wish we had used this URL for the other searches. Maybe we should update

davidsbailey replied:
there is still time! since this is only used by our internal team / on levelbuilder, it should be easy to change those urls later, too.

dmcavoy reviewed (approved):
This looks great! Nice work to get this done ahead of next week.

I think there are some things that could be cleaned up such as the framework list but that can be follow up work

dmcavoy commented:
@davidsbailey In an effort to help you get this in today I rebased this back to staging and fixed the test that was failing so that hopefully by the time you get up it has a green run

davidsbailey commented:
> @davidsbailey In an effort to help you get this in today I rebased this back to staging and fixed the test that was failing so that hopefully by the time you get up it has a green run

oh, wow! thank you, Dani!

davidsbailey commented:
merging now that the pending test run passed unit tests, given that a previous run passed ui tests, and there have been only changes to unit test code since then.
